CppCoreGuidelines.md

```diff
@@ -6903,43 +6903,57 @@ not using this (over)general interface in favor of a particular interface found
 ##### Reason
 
 A class is more stable (less brittle) if it does not contain data.
-Interfaces should normally be composed entirely of public pure virtual functions and a default/empty virtual destructor.
+An interface should be composed entirely of pure virtual functions and a virtual destructor; see [C.35](#Rc-dtor-virtual).
+An interface should have a defaulted default constructor.
```

Comment:
Arthur, can you elaborate on why to add that it should have a defaulted default constructor?

Reply:
Editors call: Below it mentions that then derived classes don't need to write explicit constructors, but that's an implementation detail and not part of the basic pattern. Could you remove the part about constructors from this PR? Thanks!

Reply:
@hsutter: It sounds like you found the part that answered your question. :)  Some people like to `=delete` the move and copy members of a polymorphic class, just to drive home that polymorphic classes must never be moved or copied. If you `=delete` any constructor, then you've user-declared that constructor, and therefore you don't get any implicitly declared default constructor — at which point your base class is useless (see https://godbolt.org/z/UjH-MJ ).

"Ah, but what if I _also_ provide a user-defined constructor such as `explicit MyBase(int)`?" — Well, what purpose does that `int` serve? It can't be used to initialize any member data, because abstract base classes should not have member data. Therefore the _only_ kind of constructor that makes sense for an abstract base class is a default constructor — and you might as well explicitly default it — and if you delete your copy operations without defaulting your default constructor, you have a bug.

TLDR: Your base class _must_ either `=default` its default constructor, _or_ not-`=delete` its copy operations. If you delete your copy operations without defaulting your default constructor, you have a bug.

Personally I would accept "Never `=delete` a polymorphic type's copy operations (but make sure never to call them either)" as a guideline, but I think realistically that would be super controversial. So I decided to stay silent on the `=delete` question, but add "always `=default` your base class's default constructor" so that this guideline would work for both people who _do_ `=delete` and people (like me) who don't.